app/models/reviewer_list.rb

```diff
@@ -22,10 +22,29 @@ def each(&block)
       .each(&block)
   end
 
-  def choose_reviewer(exclude_list: [])
-    each
-      .reject { |r| exclude_list.include?(r.login) }
-      .sample
+  def choose_reviewer(pull_request, exclude_list: [])
+    filtered_reviewers = each
+                           .reject { |r| exclude_list.include?(r.login) }
+    # Average the load from last 7 days. Irrespective of the whether it is already reviewed or pending
+    reviewers_load = Reviewer.where("login IN (?) AND created_at > ?", filtered_reviewers.map(&:login), 1.week.ago).group(:login).count
+
+    reviewers_with_no_load = filtered_reviewers.map(&:login) - reviewers_load.keys
+    reviewer = nil
+
+    if reviewers_with_no_load.present?
+      reviewer = reviewers_with_no_load.first
+    else
+      # Do not assign any reviewer from the last PR's reviewer list to avoid overburdening of any reviewer who was unavailable/leave in last 7 days.
+      last_pr_reviewers = PullRequest.where("number != ? AND repository_id = ? AND status != '#{PullRequest::STATUS_CLOSED}'",
+                                            pull_request.number, pull_request.repository_id).last&.reviewers&.map(&:login)
```

Comment:
On this query too, since it is reasonably complex, let's extract it into a query object and write tests specifically for it.

I would also prefer to use the ActiveRecord DSL as much as possible before dropping into raw SQL. You should group the two `!=` together and use `where.not`

```suggestion
      last_pr_reviewers = PullRequest.where(repository_id: pull_request.repository_id).where.not(number: pull_request.number, status: PullRequest::STATUS_CLOSED).last&.reviewers&.map(&:login)
```